cmd/gardener-extension-shoot-flux/app/options.go

```diff
@@ -0,0 +1,154 @@
+	restOptions        *extensionscmdcontroller.RESTOptions
+	managerOptions     *extensionscmdcontroller.ManagerOptions
+	extensionOptions   *extensionscmdcontroller.ControllerOptions
+	healthOptions      *extensionscmdcontroller.ControllerOptions
+	heartbeatOptions   *extensionsheartbeatcmd.Options
+	controllerSwitches *extensionscmdcontroller.SwitchOptions
+	reconcileOptions   *extensionscmdcontroller.ReconcilerOptions
+	optionAggregator   extensionscmdcontroller.OptionAggregator
+
+	// completed options
+	RESTConfig     *rest.Config
+	ManagerOptions manager.Options
+}
+
+// newOptions creates a new options instance.
+func newOptions() *options {
+	opts := &options{
+		generalOptions: &extensionscmdcontroller.GeneralOptions{},
+		restOptions:    &extensionscmdcontroller.RESTOptions{},
+		managerOptions: &extensionscmdcontroller.ManagerOptions{
+			// These are default values.
+			LeaderElection:          true,
+			LeaderElectionID:        extensionscmdcontroller.LeaderElectionNameID(fluxv1alpha1.ExtensionType),
+			LeaderElectionNamespace: os.Getenv("LEADER_ELECTION_NAMESPACE"),
+			MetricsBindAddress:      ":8080",
+			HealthBindAddress:       ":8081",
+		},
+		extensionOptions: &extensionscmdcontroller.ControllerOptions{
+			// This is a default value.
+			MaxConcurrentReconciles: 5,
+		},
+		healthOptions: &extensionscmdcontroller.ControllerOptions{
+			// This is a default value.
+			MaxConcurrentReconciles: 5,
+		},
+		heartbeatOptions: &extensionsheartbeatcmd.Options{
+			ExtensionName: fluxv1alpha1.ExtensionType,
+			Namespace:     os.Getenv("LEADER_ELECTION_NAMESPACE"),
+		},
+		controllerSwitches: extensionscmdcontroller.NewSwitchOptions(
+			extensionscmdcontroller.Switch(extension.ControllerName, extension.AddToManager),
+			extensionscmdcontroller.Switch(extensionshealthcheckcontroller.ControllerName, healthcheck.AddToManager),
+			extensionscmdcontroller.Switch(extensionsheartbeatcontroller.ControllerName, extensionsheartbeatcontroller.AddToManager),
+		),
+		reconcileOptions: &extensionscmdcontroller.ReconcilerOptions{},
+	}
+
+	opts.optionAggregator = extensionscmdcontroller.NewOptionAggregator(
+		opts.generalOptions,
+		opts.restOptions,
+		opts.managerOptions,
+		extensionscmdcontroller.PrefixOption(extension.ControllerName+"-", opts.extensionOptions),
+		extensionscmdcontroller.PrefixOption(extensionshealthcheckcontroller.ControllerName+"-", opts.healthOptions),
+		extensionscmdcontroller.PrefixOption(extensionsheartbeatcontroller.ControllerName+"-", opts.heartbeatOptions),
+		opts.controllerSwitches,
+		opts.reconcileOptions,
+	)
+
+	return opts
+}
+
+func (o *options) addFlags(fs *pflag.FlagSet) {
+	o.optionAggregator.AddFlags(fs)
+}
+
+func (o *options) Complete() error {
+	if err := o.optionAggregator.Complete(); err != nil {
+		return err
+	}
+
+	// customize rest config
+	o.RESTConfig = o.restOptions.Completed().Config
+	// TODO: consider dropping this or make these settings configurable
```

Comment:
I was wondering - there are no like well known flags / chart values for this in the gardener ecosystem, right? 
And do we even need to increase the defaults here in our case? The extension is installed per-shoot, so will never do many requests, right?